crates/cast/src/lib.rs

```diff
@@ -238,56 +246,33 @@ pub async fn wait_for_tx(
     retries: u8,
 ) -> Result<&str> {
     println!("Transaction hash: {tx_hash:#x}");
-
-    let mut maybe_receipt: Option<MaybePendingTransactionReceipt> = None;
     for i in (1..retries).rev() {
         match provider.get_transaction_receipt(tx_hash).await {
-            Ok(receipt) => {
-                maybe_receipt = Some(receipt);
-                break;
-            }
+            Ok(receipt) => match receipt.execution_result() {
+                ExecutionResult::Succeeded => {
+                    return Ok("Transaction accepted");
+                }
+                ExecutionResult::Reverted { reason } => {
+                    return Err(anyhow!("Transaction has been reverted: {}", reason));
+                }
+            },
             Err(ProviderError::StarknetError(StarknetErrorWithMessage {
                 code: MaybeUnknownErrorCode::Known(StarknetError::TransactionHashNotFound),
                 message: _,
             })) => {
-                println!("Waiting for transaction to be received. Retries left: {i}");
+                println!("Waiting for transaction to be received ({i} retries left)");
             }
             Err(err) => return Err(err.into()),
         };
 
-        sleep(Duration::from_secs(5));
-    }
-
-    if maybe_receipt.is_none() {
-        bail!("Could not get transaction with hash: {tx_hash:#x}. Transaction rejected or not received.")
-    }
-
-    loop {
-        let status = if let Ok(Receipt(receipt)) = provider.get_transaction_receipt(tx_hash).await {
-            match receipt {
-                Invoke(receipt) => receipt.status,
-                Declare(receipt) => receipt.status,
-                Deploy(receipt) => receipt.status,
-                DeployAccount(receipt) => receipt.status,
-                L1Handler(receipt) => receipt.status,
-            }
-        } else {
-            println!("Received transaction. Status: Pending");
+        if i > 1 {
             sleep(Duration::from_secs(5));
```

Comment:
I think I don't understand this `if`. Why does sleep happen only for `i > 1`?

Reply:
Ahhhh yes I am big dzban and forgot to remove this `if`, I used it for troubleshooting during testing phase. Sorry for that, and good catch :+1:.